src/com.unity.editor.tasks/Editor/Schedulers/MainThreadSynchronizationContext.cs

```diff
@@ -24,10 +34,23 @@ public override void Post(SendOrPostCallback d, object state)
 			if (d == null)
 				return;
 
-			EditorApplication.delayCall += () => d(state);
+			m_Callbacks.Enqueue(() => d(state));
 		}
 
-		public void Dispose() {}
+		void Update()
+		{
+			while (m_Callbacks.TryDequeue(out var callback))
+			{
+				try
+				{
+					callback();
+				}
+				catch
+				{
+					// Ignore exceptions.
```

Comment:
The only thing I'm wondering here is if this will mean exceptions in the UI thread won't be reported as normal to the console...